src/dlsym_default.c

```diff
@@ -345,10 +412,30 @@ if (default_symbol_index) {
   if (!default_symbol_index) {
     printf("ERROR:  No default symbol version found for %s.\n"
            "        Extend code to look for hidden symbols?\n", symbol);
-  }
-  if (default_symbol_index)
-    return tags.base_addr + tags.symtab[default_symbol_index].st_value;
-  else
     assert(0);
```

Comment:
Remove the assert. This function is only used for some debug prints. It's better to expose the fact that the default symbol could not be found to the user of dlsym_default than asserting because of some internal error.